python/lancedb/table.py

```diff
@@ -500,6 +501,65 @@ def add(
         lance.write_dataset(data, self._dataset_uri, schema=self.schema, mode=mode)
         self._reset_dataset()
 
+    def merge(
+        self,
+        other_table: Union[LanceTable, ReaderLike],
+        left_on: str,
+        right_on: Optional[str] = None,
+        schema: Optional[pa.Schema, LanceModel] = None,
+    ):
+        """Merge another table into this table.
+
+        Performs a left join, where the dataset is the left side and data_obj
+        is the right side. Rows existing in the dataset but not on the left will
+        be filled with null values, unless Lance doesn't support null values for
+        some types, in which case an error will be raised.
```

Comment:
```suggestion
        is the right side. Rows existing in the dataset that do not match any value
        in other_table will be filled with null values, unless Lance doesn't support
        null values for some types, in which case an error will be raised.
```

Which types don't support null values?

Reply:
Only String and List support nulls right now. All others don't support them.